### ISSUE TITLE
Intermittent no response with many concurrent requests

### ISSUE THREAD

**Issue by jonasfj** [inconshreveable/ngrok#105]
I have a project where I include some 60 files (I'll merge and minify many of them for production), but during development it makes a lot of sense to keep them.

In general ngrok works fine, but when I load a page that does 60 concurrent requests one of them remains pending and after 60 seconds the browser does a `504` timeout.
This happens about 50% of the time. For a page with only 10 includes it only happens if I repeated reload it very fast.

I suspect that the connection that fails isn't joined and just times out. Perhaps, retrying `c.out <- &msg.ReqProxy{}` in [control.go line 325]( https://github.com/inconshreveable/ngrok/blob/master/src/ngrok/server/control.go#L325) would fix it... I don't know... But from what I can see, it seems the connection is just dropped somewhere. And with 60 concurrent requests, there is competition over who gets the next connection from client.

According the my server log and ngrok request log, the request that fails doesn't seem to reach my local computer. So it's probably dropped in ngrokd.

The response is in fact:
```html
<html>
<head><title>504 Gateway Time-out</title></head>
<body bgcolor="white">
<center><h1>504 Gateway Time-out</h1></center>
<hr><center>nginx/1.4.6</center>
</body>
</html>
```

I've attached a log here: https://gist.github.com/jonasfj/84ea399eb26b0e38ad0d

**Comment by inconshreveable** [comment]
this problem is fixed in 2.x. won't fix for 1.x

**Comment by tim-massey** [comment]
Not fixed in `v2.1.3` for me
